packages/pricing/src/services/pricing-module.ts

```diff
@@ -79,46 +96,91 @@ export default class PricingModuleService<
     pricingContext: PricingContext = { context: {} },
     @MedusaContext() sharedContext: Context = {}
   ): Promise<PricingTypes.CalculatedPriceSetDTO> {
+    const manager = sharedContext.manager as EntityManager
```

Comment:
todo: I believe that we should eventually create a PricingModuleRepository or the baseRepo you can decide the best approach, and put that in it to separate the responsibility of accessing the db directly

Reply:
can you elaborate on this? I don't quite understand